frontend/src/components/object/ObjectVersion.vue

```diff
@@ -21,72 +20,46 @@ type VersionDataSource = {
 type Props = {
   bucketId: string;
   objectId: string;
-  versionId: string;
 };
 
 const props = withDefaults(defineProps<Props>(), {});
 
+// Emits
+const emit = defineEmits(['on-deleted-success']);
+
 // Store
 const permissionStore = usePermissionStore();
 const userStore = useUserStore();
 const versionStore = useVersionStore();
+// getters
 const { getUserId } = storeToRefs(useAuthStore());
-const { getUserSearch } = storeToRefs(userStore);
-const { getVersions } = storeToRefs(versionStore);
+const { getUser } = storeToRefs(userStore);
+const { getVersionsByObjectId } = storeToRefs(versionStore);
 
 // State
-const tableData: Ref<Array<VersionDataSource>> = ref([]);
-
-// Actions
-const router = useRouter();
+const versionId = defineModel<string>('versionId');
+const versions: Ref<Array<Version>> = computed((): any => getVersionsByObjectId.value(props.objectId));
+const tableData: Ref<Array<VersionDataSource>> = computed(() => {
+  return versions.value.map((v: Version, index, arr) => ({
+    ...v,
+    createdByName: getUser.value(v.createdBy)?.fullName,
+    versionNumber: arr.length - index
+  }));
+});
 
 // Highlight row for currently selected version
-const rowClass = (data: any) => [{ 'selected-row': data.id === props.versionId }];
-
-async function onDeletedSuccess(versionId: string) {
-  await versionStore.fetchVersions({ objectId: props.objectId });
+const rowClass = (data: any) => [{ 'selected-row': data.id === versionId.value }];
 
-  // Navigate to new latest version if deleting active version
-  if (props.versionId === versionId) {
-    router.push({
-      name: RouteNames.DETAIL_OBJECTS,
-      query: {
-        objectId: props.objectId,
-        versionId: versionStore.findLatestVersionIdByObjectId(props.objectId)
-      }
-    });
-  }
+async function onDeletedSuccess() {
+  emit('on-deleted-success');
 }
 
-const rowClick = function (e: any) {
-  router.push({
-    name: RouteNames.DETAIL_OBJECTS,
-    query: { objectId: e.data.objectId, versionId: e.data.id }
-  });
+const onVersionClick = (e: any) => {
+  versionId.value = e.data.id;
 };
 
-async function load() {
-  await versionStore.fetchVersions({ objectId: props.objectId });
-  const versions = versionStore.findVersionsByObjectId(props.objectId);
-  await userStore.fetchUsers({ userId: versions.map((x: Version) => x.createdBy) });
-}
-
-onMounted(() => {
-  load();
-});
-
-watch(props, () => {
-  load();
-});
-
-watch(getVersions, async () => {
-  const versions = versionStore.findVersionsByObjectId(props.objectId);
-  await userStore.fetchUsers({ userId: versions.map((x: Version) => x.createdBy) });
-  tableData.value = versions.map((v: Version, index, arr) => ({
-    ...v,
-    createdByName: getUserSearch.value.find((u: User) => u.userId === v.createdBy)?.fullName,
-    versionNumber: arr.length - index
-  }));
+watch(props, async () => {
+  await userStore.fetchUsers({ userId: versions.value.map((x: Version) => x.createdBy) });
```

Comment:
Not completely certain if we really need the async and await here as this callback just fires and finishes when it finishes. There doesn't appear to be any dependent functions that need to be blocking on this specific thread.